text/0000-pack-direct-podman-call.md

```diff
@@ -0,0 +1,86 @@
+# Meta
+[meta]: #meta
+- Name: Pack direct podman call
+- Start Date: 2023-06-16
+- Author(s): dvaumoron
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+A flag `--container-engine` will be added to the `pack` CLI, the allowed flag value will be `docker` (the current functioning) and `podman` to call podman directly (without socket call)
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+This will allow to use pack as a standalone on the OS supported by podman (currently Linux, as Windows and macOS need podman machine), making possible to use pack without installing docker or podman (a tool like skopeo will still be necessary to send image in registries)
+
+# What it is
+[what-it-is]: #what-it-is
+
+Adding `--container-engine podman` to the `pack` command which currently need docker call would allow them to work without docker installed.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The flag will change the initialization of the CommonAPIClient passed to call docker (interface defined in "github.com/docker/docker/client") by an adapter conforming to the used subset [DockerClient](https://github.com/buildpacks/pack/blob/v0.29.0/pkg/client/docker.go#L14) to call podman as a library (forwarding calls to an initialized instance of [Runtime](https://github.com/containers/podman/blob/v4.5.1/libpod/runtime.go#L63)).
+
+# Migration
+[migration]: #migration
+
+N/A
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+pack contributor could need to update the podman dependencies in go.mod for bugfixes
```

Comment:
Does the `podman` API support running `arm64` containers on `amd64`?  If not, then one drawback is that this proposed RFC will not benefit from changes in our multi-architecture RFC.

Reply:
This seems to be ok : https://github.com/containers/podman/discussions/15741
podman use the --arch flag which is transmited to the interface ContainerEngine as an option (i changed the how it works part to change Runtime by ContainerEngine (the Runtime struct is called by one of the ContainerEngine implementation))